<launch>
    <!-- SCENE : 2.5D Office -->
    <arg name="fov_h" value="70.0"/>
    <arg name="fov_w" value="85.0"/>
    <arg name="max_dist" value="3.2"/>
    <arg name="resolution_" value="0.2"/>

    <node pkg ="viewpoint_manager" name ="viewpoint_manager" type ="viewpoint_manager_node" output = "screen" required ="true">

        <!-- ! Viewpoint Manager Node ! -->
        <param name="viewpoint_manager_node/cloud_path" value="$(find viewpoint_manager)/resource/office_surface.pcd" type="string"/>
        <param name="viewpoint_manager_node/free_cloud_path" value="$(find viewpoint_manager)/resource/office_free.pcd" type="string"/>
        <param name="viewpoint_manager_node/downsampled_size" value="0.3" type="double"/>
        <param name="viewpoint_manager_node/min_box_z" value="1.5" type="double"/>
        <param name="viewpoint_manager_node/max_box_z" value="2.5" type="double"/>
        <!-- <param name="viewpoint_manager_node/downsampled_size_for_viewpoint" value="0.3" type="double"/> -->
        <param name="viewpoint_manager_node/downsampled_size_for_viewpoint" value="0.35" type="double"/>
        <!-- <param name="viewpoint_manager_node/downsampled_size_for_viewpoint" value="0.4" type="double"/> -->
        <!-- <param name="viewpoint_manager_node/downsampled_size_for_viewpoint" value="0.46" type="double"/> -->
        <!-- <param name="viewpoint_manager_node/downsampled_size_for_viewpoint" value="0.5" type="double"/> -->

        <!-- ! Mapping ! -->
        <param name="hcmap/resolution" value="$(arg resolution_)" type="double"/>
        <param name="hcmap/interval" value="1.0" type="double"/>
        <param name="hcmap/plane_thickness" value="3.0" type="double"/>
        <param name="hcmap/dilateRadius" value="0.5" type="double"/>
        <param name="hcmap/checkScale" value="5.0" type="double"/>
        <param name="hcmap/checkSize" value="7" type="int"/>
        <param name="hcmap/inflateVoxel" value="4" type="int"/>

        <!-- ! Perception ! -->
        <param name="perception_utils/top_angle" value="$(eval arg('fov_h')*3.1415926/360.0)" type="double"/>
        <param name="perception_utils/left_angle" value="$(eval arg('fov_w')*3.1415926/360.0)" type="double"/>
        <param name="perception_utils/right_angle" value="$(eval arg('fov_w')*3.1415926/360.0)" type="double"/>
        <param name="perception_utils/max_dist" value="$(arg max_dist)" type="double"/>
        <param name="perception_utils/vis_dist" value="0.9" type="double"/>

        <!-- ! Viewpoint Manager ! -->
        <param name="viewpoint_manager/visible_range" value="$(arg max_dist)" type="double"/>
        <param name="viewpoint_manager/viewpoints_distance" value="2.5" type="double"/>
        <param name="viewpoint_manager/fov_h" value="$(arg fov_h)" type="double"/>
        <param name="viewpoint_manager/fov_w" value="$(arg fov_w)" type="double"/>
        <param name="viewpoint_manager/pitch_upper" value="70.0" type="double"/>
        <param name="viewpoint_manager/pitch_lower" value="-85.0" type="double"/>
        <param name="viewpoint_manager/zGround" value="true" type="bool"/>
        <param name="viewpoint_manager/GroundPos" value="0.0" type="double"/>
        <param name="viewpoint_manager/safeHeight" value="0.5" type="double"/>
        <param name="viewpoint_manager/safe_radius" value="0.4" type="double"/>
        <param name="viewpoint_manager/attitude_type" value="yaw" type="string"/>   <!-- attitude_type : 1) yaw 2) all -> (pitch, yaw) -->
        <param name="viewpoint_manager/max_iter_num" value="3" type="int"/>
        <param name="viewpoint_manager/pose_update" value="false" type="bool"/>
    </node>

</launch>